mlir/tools/mlir-miopen-lib/cmake/MLIRMIOpenConfig.cmake

```diff
@@ -0,0 +1,4 @@
+# Auto compute the include dir for the header file
+get_filename_component(SELF_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+include("${SELF_DIR}/MLIRMIOpenTargets.cmake")
+get_filename_component(MLIRMIOpen_INCLUDE_DIR "${SELF_DIR}/../../../include/@PACKAGE_DIR@" ABSOLUTE)
```

Comment:
What's the last line here for?

Reply:
It sets the include_dir of Miir.h so that MIOpen should be able to include it like
`target_include_directories(MIOpen SYSTEM PRIVATE ${MLIRMIOpen_INCLUDE_DIR}) `

Reply:
Is that the usual way other CMake packages handle this?
And shouldn't we be exporting the path to the library as well?

Reply:
The exported library is loaded by add_library as
`add_library(MLIR::MLIRMIOpen STATIC IMPORTED)` when find_package() locates and loads the packageConfig.cmake file. The path to the library is also set as a property of the target as `IMPORTED_LOCATION_CONFIG`.

In the config mode of find_package(), no variable is set to hold the include dir. Therefore, we need to set the INCLUDE_DIR manually. 
(I am new to cmake and please correct me if I mis-understood anything)

Reply:
It looks like https://cmake.org/cmake/help/v3.15/module/CMakePackageConfigHelpers.html might be of interest, so as to make the whole installation setup more general?